File: 2Lab/labs4/lab4.cs
using System;


class StringBase
{
    protected string data;

    public StringBase()
    {
        data = "";
    }

    public StringBase(string str)
    {
        data = str;
    }

    public StringBase(StringBase other)
    {
        data = other.data;
    }

    public string GetData()
    {
        return data;
    }

    public int GetLength()
    {
        return data.Length;
    }
}

class LetterString : StringBase
{
    public LetterString() : base() { }

    public LetterString(string str) : base(str) { }

    public LetterString(LetterString other) : base(other) { }

    public string GetSorted()
    {
        char[] chars = data.ToCharArray();
        Array.Sort(chars);
        return new string(chars);
    }
}

// Main entry point
class Program
{
    static void Main()
    {
        LetterString ls = new LetterString("helloWorld");

        Console.WriteLine("String: " + ls.GetData());
        Console.WriteLine("Length: " + ls.GetLength());
        Console.WriteLine("Sorted: " + ls.GetSorted());
    }
}

File: 2Lab/labs4/lab4.cpp
#include <iostream>
#include <string>
#include <algorithm>


class StringBase {
protected:
    std::string data;

public:
    StringBase() : data("") {}
    StringBase(const std::string& str) : data(str) {}
    StringBase(const StringBase& other) : data(other.data) {}
    virtual ~StringBase() {}

    std::string getData() const {
        return data;
    }

    size_t getLength() const {
        return data.length();
    }
};


class LetterString : public StringBase {
public:
    LetterString() : StringBase() {}
    LetterString(const std::string& str) : StringBase(str) {}
    LetterString(const LetterString& other) : StringBase(other) {}
    ~LetterString() {}

    std::string getSorted() const {
        std::string sorted = data;
        std::sort(sorted.begin(), sorted.end());
        return sorted;
    }
};


int main() {
    LetterString ls("helloWorld");

    std::cout << "String: " << ls.getData() << std::endl;
    std::cout << "Length: " << ls.getLength() << std::endl;
    std::cout << "Sorted: " << ls.getSorted() << std::endl;

    return 0;
}
